### PR TITLE
(GH-580) Add target framework versions to extension metadata

### PR DESCRIPTION
**Pull request by pascalberger**
Add support for target framework versions to extension metadata.

Part of #580

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -48,6 +48,7 @@ class ExtensionSpec
     public List<string> Categories { get; set; }
     public string AnalyzedPackageVersion { get; set; }
     public string TargetCakeVersion { get; set; }
+    public List<string> TargetFrameworkVersions { get; set; }
```
Comment (Jericho) [inline build.cake:51]: Today I learned that TFM stands for "Target Framework Moniker" ([here](https://docs.microsoft.com/en-us/dotnet/standard/frameworks)). So how about we name this property `TargetFrameworkMonikers`?

**Review by augustoproiete — APPROVED** [review]

**Review by Jericho — CHANGES_REQUESTED** [review]
Here's a suggestion:

File: build.cake (hunk 1 above)
Comment (augustoproiete) [reply build.cake:51]: `TargetFrameworkMonikers` works for me, as well as `TargetFrameworks` which is what holds the TFMs in the build.

```
  <PropertyGroup>
    <TargetFrameworks>net5.0;netstandard2.0;net461</TargetFrameworks>
  </PropertyGroup>
```

Comment (Jericho) [reply build.cake:51]: `TargetFrameworks` is shorter, I like it.

Comment (pascalberger) [reply build.cake:51]: Works both for me. How about going with `TargetFrameworks`?

Comment (Jericho) [reply build.cake:51]: I vote for `TargetFrameworks`. It's shorter, it's similar to the XML node in.csproj files. It's better than my suggestion.

Comment (pascalberger) [reply build.cake]: Updated to `TargetFrameworks`

**Review by Jericho — APPROVED** [review]
Looks good!

**Pull request merged**